imageprocessing/src/main/java/com/example/imageprocessing/PreProcessing.java

```diff
@@ -0,0 +1,394 @@
+     */
+    private Bitmap editBright(Bitmap image){
+        /*This variable is used to put a limit to the change of the image's brightness.
+          The value 240 is derived from the fact that in the for loop we try to modify
+          the value of all the pixels of a step, and being the maximum value = 255 (pixel's
+          color maximum value, we put the limit on 240*/
+        final int maxBrightness = 240;
+        final int step = 15;
+
+        //Converts the image into a matrix
+        Mat bright;
+        try{
+            bright = IPUtils.conversionBitmapToMat(image);
+        } catch (ConversionFailedException error){
+            Log.e(TAG, error.getErrorMessage());
+            return image;
+        }
+
+        //Call the internal method isBright to detect if the image is bright or dark
+        //and change the brightness according to the number obtained
+        Mat modifiedMat = new Mat();
+        switch (isBright(bright)) {
+            case IMAGE_TOO_BRIGHT:
+                Log.d(TAG, "Case==IMAGE_TOO_BRIGHT");
+                //Darkens the colour's brightness until it's in an optimal value
+                for(double changeBrightness = 0; changeBrightness != maxBrightness;
+                    changeBrightness -= step){
+                   /*This variable is used to select the type of matrix we want to abtain in the
+                      the convertTo method. If it's negative the type doesn't change*/
+                    int matrixType = -1;
+
+                    /*This variable is used to change the contrast of the matrix, but we want
+                      only modify the brightness so we put the value 1 because the method use
+                      this formula from the documentation:
+                      m(x,y) = saturate _ cast<rType>(alpha(*this)(x,y) + beta)
+                      We called beta as changeBrightness*/
+                    int alpha = 1;
+
+                    /*Modify the values of all pixels with an alpha and beta value following
+                      the formula above.*/
+                    bright.convertTo(modifiedMat, matrixType, alpha, changeBrightness);
+
+                    //Verify if the image is good enough
+                    if(isBright(modifiedMat) == BrightnessValue.IMAGE_IS_OK){
+                        try{
+                            return IPUtils.conversionMatToBitmap(modifiedMat);
+                        } catch (ConversionFailedException error){
+                            Log.e(TAG, error.getErrorMessage());
+                            return image;
+                        }
+                    }
+                }
+                break;
+
+            case IMAGE_TOO_DARK:
+                Log.d(TAG, "Case==IMAGE_TOO_DARK");
+                //The variables are explained in the case above
+                //Lightens the colour's brightness until it's in an optimal value
+                for(double changeBrightness = 0; changeBrightness != maxBrightness;
```

Comment:
Maybe I misunderstood also here but In this way I can end to an infinite loop (avoided also here by
if(isBright(modifiedMat) == BrightnessValue.IMAGE_IS_OK)).
I suggest to use
for(double changeBrightness = 0; changeBrightness < maxBrightness; changeBrightness += step) 
What do you think?

Reply:
We are making the switch smaller and avoiding the for cycles